samples/extensions/vertex_dynamic_state/vertex_dynamic_state.cpp

```diff
@@ -0,0 +1,634 @@
+/* Copyright (c) 2022, Mobica Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "vertex_dynamic_state.h"
+
+vertex_dynamic_state::vertex_dynamic_state()
+{
+	title = "Vertex Dynamic State";
+
+	add_instance_extension(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME);
+	add_device_extension(VK_EXT_VERTEX_INPUT_DYNAMIC_STATE_EXTENSION_NAME);
+}
+
+vertex_dynamic_state::~vertex_dynamic_state()
+{
+	if (device)
+	{
+		vkDestroySampler(get_device().get_handle(), textures.envmap.sampler, VK_NULL_HANDLE);
+		textures = {};
+		skybox.reset();
+		object.reset();
+		ubo.reset();
+
+		vkDestroyPipeline(get_device().get_handle(), model_pipeline, VK_NULL_HANDLE);
+		vkDestroyPipeline(get_device().get_handle(), skybox_pipeline, VK_NULL_HANDLE);
+		vkDestroyPipelineLayout(get_device().get_handle(), pipeline_layout, VK_NULL_HANDLE);
+		vkDestroyDescriptorSetLayout(get_device().get_handle(), descriptor_set_layout, VK_NULL_HANDLE);
+		vkDestroyDescriptorPool(get_device().get_handle(), descriptor_pool, VK_NULL_HANDLE);
+	}
+}
+
+/**
+ * 	@fn bool vertex_dynamic_state::prepare(vkb::Platform &platform)
+ * 	@brief Configuring all sample specific settings, creating descriptor sets/pool, pipelines, generating or loading models etc. 
+*/
+bool vertex_dynamic_state::prepare(vkb::Platform &platform)
+{
+	if (!ApiVulkanSample::prepare(platform))
+	{
+		return false;
+	}
+
+#if VK_NO_PROTOTYPES
+	VkInstance instance = get_device().get_gpu().get_instance().get_handle();
+	assert(!!instance);
+	vkCmdSetVertexInputEXT = (PFN_vkCmdSetVertexInputEXT) vkGetInstanceProcAddr(instance, "vkCmdSetVertexInputEXT");
+	if (!vkCmdSetVertexInputEXT)
+	{
+		throw std::runtime_error("Unable to dynamically load vkCmdSetVertexInputEXT");
+	}
+#endif
```

Comment:
> Why do you need to get that function pointer? Doesn't volk resolve that function?

Our understanding of that concept was, that this is needed if vulkan is linked dynamically. Since in vulkan_core.h you can see:
```cpp
#ifndef VK_NO_PROTOTYPES
VKAPI_ATTR void VKAPI_CALL vkCmdSetVertexInputEXT(
    VkCommandBuffer                             commandBuffer,
    uint32_t                                    vertexBindingDescriptionCount,
    const VkVertexInputBindingDescription2EXT*  pVertexBindingDescriptions,
    uint32_t                                    vertexAttributeDescriptionCount,
    const VkVertexInputAttributeDescription2EXT* pVertexAttributeDescriptions);
#endif
```
We based this solution on dynamic_rendering sample where `vkCmdBeginRenderingKHR` and `vkCmdEndRenderingKHR` are resolved the same way.

Without this our code did not compile. Let us know if this is correct approach.

Reply:
Ah, I see... the problem is the volk version used here.
If you update it to a more recent one, you don't need to explicitly get the function pointer on your own.

Reply:
Okay, the solution is easy in the case of implementation, but updating third party library will impact all created samples in this project. This kind of change needs to be discussed with other reviewers/contributors connected to this project. For now, we can't just update the code, but on Monday as I know will be Khronos Samples meeting. We can introduce the possibility of that kind of upgrade.

Reply:
@TomAtkinsonArm @SaschaWillems @gary-sweet Can we discuss the possibility to implement that kind of change in Khronos Samples project? I didn't know if we have some regulations to stay with current versions of third-party libraries. Any feedback regarding this topic will be helpful to me.

In nutshell: the idea is to update `volk` library to a newer version so that we don't need to add this kind of code declaration in our Samples (see code pointed by @asuessenbach).

Reply:
Volk is included as a submodule. So usually if a sample is added that needs an update to Volk, the PR for that sample also updates that submodule.